src/page-modules/assistant/client/journey-planner/index.ts

```diff
@@ -1,18 +1,55 @@
-import { TripData } from '../../server/journey-planner/validators';
-import { TripQuery } from '../../types';
+import useSWR from 'swr';
+import { NonTransitTripData, TripQuery, TripData } from '../../types';
+import { swrFetcher } from '@atb/modules/api-browser';
+import useSWRInfinite from 'swr/infinite';
 
 export type TripApiReturnType = TripData;
 
-export async function nextTripPatterns(
-  query: TripQuery,
-): Promise<TripApiReturnType> {
-  const queryString = tripQueryToQueryString(query);
+const getKey =
+  (query: TripQuery) =>
+  (pageIndex: number, previousPageData: TripApiReturnType) => {
```

Comment:
Generally unsure if this is the most readable way to do "double functions". What you are doing here is essentially allowing for [partial application](https://en.wikipedia.org/wiki/Partial_application). But even without doing structured partial application i think maybe this would be easier to read without having two arrow functions? 🤔

```ts
function createKeyGetter(query: TripQuery) {
  return (pageIndex: number, previousPageData: TripApiReturnType) => {
```

Reply:
I didn't think of that. I agree, more readable for sure